ludwig/benchmarking/summary_dataclasses.py

```diff
@@ -66,20 +86,200 @@ class ExperimentsDiff:
+
+                f.write(
+                    csv_str.format(
+                        self.dataset_name,
+                        output_feature_name,
+                        metric_name,
+                        experiment1_val,
+                        experiment2_val,
+                        diff,
+                        diff_percentage,
+                    )
+                )
+        print("Exported report to", full_path)
+
+    def __str__(self):
+        ret = []
+        spacing_str = "{:<20} {:<23} {:<13} {:<13} {:<13} {:<5}"
+        ret.append(f"Metrics for dataset: {self.dataset_name}\n")
+        ret.append(
+            spacing_str.format(
+                "Output Feature Name",
+                "Metric Name",
+                self.base_experiment_name,
+                self.experimental_experiment_name,
+                "Diff",
+                "Diff Percentage",
+            )
+        )
+        for metric in sorted(self.metrics, key=lambda m: m.name):
+            output_feature_name = self.base_summary.output_feature_name
+            metric_name = metric.name
+            experiment1_val = round(metric.base_value, 3)
+            experiment2_val = round(metric.experimental_value, 3)
+            diff = round(metric.diff, 3)
+            diff_percentage = metric.diff_percentage
+            if isinstance(diff_percentage, float):
+                diff_percentage = round(metric.diff_percentage, 3)
+            ret.append(
+                spacing_str.format(
+                    output_feature_name,
+                    metric_name,
+                    experiment1_val,
+                    experiment2_val,
+                    diff,
+                    diff_percentage,
+                )
+            )
+        return "\n".join(ret)
+
+
+@dataclass
+class ResourceUsageSummary:
+    """Summary of resource usage metrics from one experiment."""
+
+    # Path containing the JSON that stores resource usage metrics for the experiment.
+    path: str
+
+    # The tag with which the code block/function is labeled.
+    code_block_tag: str
+
+    # Dictionary that maps from metric name to their values.
+    metric_to_values: Dict[str, Union[float, int]]
+
+    # Names of metrics for the output feature.
+    metric_names: set
+
+
+@dataclass
+class ResourceUsageDiff:
+    """Store resource usage diffs for two experiments."""
 
-    # Experiment summary of the experimental experiment.
-    experimental_summary: ExperimentSummary
+    # The tag with which the code block/function is labeled.
+    code_block_tag: str
+
+    # Name of the base experiment (the one we benchmark against).
+    base_experiment_name: str
 
-    # List of metric diffs for a metric of the two experiments.
-    metrics: List[MetricDiff]
+    # Name of the experimental experiment.
+    experimental_experiment_name: str
+
+    # `List[Diff]` containing diffs for metric of the two experiments.
+    metrics: List[Diff]
+
+    def to_csv(self, path):
```

Comment:
This to_csv method probably shouldn't belong to the data class. The implementation of how the data class should be materialized, like in a file system, should be independent of the data class itself.

I'd recommend defining separate methods, i.e. `write_experiment_summary_to_csv(experiment_summary: ExperimentSummary, path)`, `write_resource_usage_summary_to_csv(resource_usage_summary: ResourceUsageSummary, path)`, parallel to the `build_*` methods that you've defined for these dataclasses.